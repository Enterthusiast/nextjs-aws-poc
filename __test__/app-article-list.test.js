/* eslint-env jest */

import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import AppArticleList from '../components/app-article-list.js'

describe('With Enzyme', () => {
  it('AppArticleList display the articles', () => {
    const props = {
      articleList : [
        {
          id: '0291cb80-5392-11e8-a5ca-e1494b1a08db',
          title: 'title test 1',
          content: 'content test 1'
        },
        {
          id: '7d6850d0-538e-11e8-a5ca-e1494b1a08db',
          title: 'title test 2',
          content: 'content test 2'
        }
      ]
    }
    const app = mount(<AppArticleList {...props}/>)

    expect(app.find('div .article-0291cb80-5392-11e8-a5ca-e1494b1a08db h2').text()).toEqual('title test 1')
    expect(app.find('div .article-7d6850d0-538e-11e8-a5ca-e1494b1a08db h2').text()).toEqual('title test 2')

  })

  it('AppArticleList do not display the articles', () => {
    const props = {
      articleList : []
    }
    const app = mount(<AppArticleList {...props}/>)

    expect(app.find('div .character-not-found')).toBeTruthy()
  })
})

describe('With Snapshot Testing', () => {
  it('AppArticleList match snapshot', () => {
    const component = renderer.create(<AppArticleList />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})