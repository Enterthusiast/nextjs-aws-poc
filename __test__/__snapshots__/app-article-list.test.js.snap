// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing AppArticleList match snapshot 1`] = `
<div
  className="sc-bdVaJa jMztiP"
>
  <div>
    <section
      className="section"
    >
      <div
        className="container"
      >
        <a
          href="/default/"
          onClick={[Function]}
        >
          &lt; Home
        </a>
        <h1
          className="title sc-bwzfXH gLzMBj"
        >
          Articles
        </h1>
        <p
          className="subtitle"
        >
          Article(s) coming from a 
          <strong>
            serverless combo
          </strong>
          , DynamoDB + AWS Lambda.
        </p>
      </div>
    </section>
    <section
      className="section is-medium"
    >
      <div
        className="container"
      >
        <div>
          No article found.
        </div>
      </div>
    </section>
  </div>
</div>
`;
